src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml

```diff
@@ -0,0 +1,168 @@
+                                <ContentPresenter Name="ContentPresenter"
+                                                  Grid.Column="1"
+                                                  Margin="{TemplateBinding Padding}"
+                                                  HorizontalAlignment="Center"
+                                                  VerticalAlignment="Center"
+                                                  Foreground="{TemplateBinding Foreground}"
+                                                  AutomationProperties.AccessibilityView="Raw"
+                                                  Content="{TemplateBinding Content}"
+                                                  ContentTemplate="{TemplateBinding ContentTemplate}"
+                                                  ContentTransitions="{TemplateBinding ContentTransitions}"/>
+                            </Grid>
+                        </ControlTemplate>
+                    </Setter.Value>
+                </Setter>
+            </Style>
+            <Style x:Key="SubTitleTextBoxStyle" TargetType="TextBlock">
+                <Setter Property="FontSize" Value="16"/>
+                <Setter Property="FontWeight" Value="SemiBold"/>
+            </Style>
+
+            <Style x:Key="ErrorTextBoxStyle" TargetType="TextBox">
+                <Setter Property="BorderBrush" Value="{ThemeResource GraphSettingsErrorBorderBrush}"/>
+                <Setter Property="Background" Value="{ThemeResource GraphSettingsErrorBackgroundBrush}"/>
+            </Style>
+        </ResourceDictionary>
+    </UserControl.Resources>
+
+    <Grid>
+        <StackPanel Margin="4,0,4,4">
+            <TextBlock x:Name="GraphOptionsHeading"
+                       x:Uid="GraphOptionsHeading"
+                       FontSize="22"
+                       AutomationProperties.HeadingLevel="Level1"/>
+            <TextBlock x:Name="GridHeading"
+                       x:Uid="GridHeading"
+                       Margin="0,12,0,6"
+                       Style="{StaticResource SubTitleTextBoxStyle}"
+                       AutomationProperties.HeadingLevel="Level2"/>
+            <Grid>
+                <Grid.RowDefinitions>
+                    <RowDefinition Height="auto"/>
+                    <RowDefinition Height="10"/>
+                    <RowDefinition Height="auto"/>
+                </Grid.RowDefinitions>
+                <Grid.ColumnDefinitions>
+                    <ColumnDefinition Width="2*"/>
+                    <ColumnDefinition Width="10"/>
+                    <ColumnDefinition Width="2*"/>
+                </Grid.ColumnDefinitions>
+
+                <TextBox x:Name="SettingsXMin"
+                         x:Uid="GraphSettingsXMin"
+                         Style="{x:Bind SelectTextBoxStyle(ViewModel.XError), Mode=OneWay}"
+                         PreviewKeyDown="GridSettingsTextBox_PreviewKeyDown"
+                         Text="{x:Bind ViewModel.XMin, Mode=TwoWay}"/>
```

Comment:
I think this needs to be two way, or have some formatting logic since the texbox accepts input like '5-44' but I am not really sure what the double value is and it never becomes updated in the UI

Reply:
I don't understand, can you precise?

Reply:
Try typing '5-44' into one of the textboxes. It is accepted and converted to some double behind the scenes, but the resulting double is never displayed in the UI.

Reply:
Fixed! std::stod will stop at the first character it doesn't know how to manage. std::stod("5-44") will stop the parsing after the '5' character for example.

I've updated to code to force stod to fully parse the string